modules/nested-elements/widgets/nested-tabs.php

```diff
@@ -0,0 +1,441 @@
+						'{{WRAPPER}}' => '--tabs-title-typography-line-height: {{SIZE}}{{UNIT}};',
+					],
+				],
+				'letter_spacing' => [
+					'selectors' => [
+						'{{WRAPPER}}' => '--tabs-title-typography-letter-spacing: {{SIZE}}{{UNIT}};',
+					],
+				],
+				'word_spacing' => [
+					'selectors' => [
+						'{{WRAPPER}}' => '--tabs-title-typography-word-spacing: {{SIZE}}{{UNIT}};',
+					],
+				],
+			],
+		] );
+
+		$this->add_group_control( Group_Control_Text_Shadow::get_type(), [
+			'name' => 'title_shadow',
+			'fields_options' => [
+				'text_shadow' => [
+					'selectors' => [
+						'{{WRAPPER}}' => '--tabs-title-shadow: {{HORIZONTAL}}px {{VERTICAL}}px {{BLUR}}px {{COLOR}};',
+					],
+				],
+			],
+		] );
+
+		$this->end_controls_section();
+	}
+
+	protected function render() {
+		$tabs = $this->get_settings_for_display( $this->get_name() );
+
+		$id_int = substr( $this->get_id_int(), 0, 3 );
+
+		$a11y_improvements_experiment = Plugin::$instance->experiments->is_feature_active( 'a11y_improvements' );
+
+		$this->add_render_attribute( 'elementor-tabs', 'class', 'elementor-tabs' );
+
+		$tabs_title_html = '';
+		$tabs_content_html = '';
+
+		foreach ( $tabs as $index => $item ) {
+			// Tabs title.
+			$tab_count = $index + 1;
+			$tab_title_setting_key = $this->get_repeater_setting_key( 'tab_title', 'tabs', $index );
+			$tab_title = $a11y_improvements_experiment ? $item['tab_title'] : '<a href="">' . $item['tab_title'] . '</a>';
+			$tab_title_mobile_setting_key = $this->get_repeater_setting_key( 'tab_title_mobile', 'tabs', $tab_count );
+
+			$this->add_render_attribute( $tab_title_setting_key, [
+				'id' => 'elementor-tab-title-' . $id_int . $tab_count,
+				'class' => [ 'elementor-tab-title', 'elementor-tab-desktop-title' ],
+				'aria-selected' => 1 === $tab_count ? 'true' : 'false',
+				'data-tab' => $tab_count,
+				'role' => 'tab',
+				'tabindex' => 1 === $tab_count ? '0' : '-1',
+				'aria-controls' => 'elementor-tab-content-' . $id_int . $tab_count,
+				'aria-expanded' => 'false',
+			] );
+
+			$this->add_render_attribute( $tab_title_mobile_setting_key, [
+				'class' => [ 'elementor-tab-title', 'elementor-tab-mobile-title' ],
+				'aria-selected' => 1 === $tab_count ? 'true' : 'false',
+				'data-tab' => $tab_count,
+				'role' => 'tab',
+				'tabindex' => 1 === $tab_count ? '0' : '-1',
+				'aria-controls' => 'elementor-tab-content-' . $id_int . $tab_count,
+				'aria-expanded' => 'false',
+			] );
+
+			$title_render_attributes = $this->get_render_attribute_string( $tab_title_setting_key );
+			$mobile_title_attributes = $this->get_render_attribute_string( $tab_title_mobile_setting_key );
+
+			$tabs_title_html .= "
+			<div $title_render_attributes >
+			$tab_title
+			</div>";
+
+			// Tabs content.
+			ob_start();
+			$this->print_children( $index );
+			$tab_content = ob_get_clean();
+
+			$tabs_content_html .= "
+			<div $mobile_title_attributes >
+				$tab_title
+			</div>
+			$tab_content
+			";
```

Comment:
Please remove all whitespaces (see the source on the frontend)

Reply:
Any hints todo it?

Reply:
Handled.